orttraining/orttraining/test/training_api/checkpoint/checkpoint_test.cc

```diff
@@ -175,7 +175,8 @@ class OrtValueTensorData {
  * Save Optimizer states into ORT checkpoint files,
  * Then load it into ORT, compare with the initial optimizer states values.
  */
-TEST(CheckpointApiTest, SaveOptimizerStateAsCheckpoint_ThenLoad_CPU) {
```

Comment:
looks we have to implement a AdamWOptimizer on CPU, for those pure CPU tests.

Reply:
Yeah. As soon as we construct the optimizer class, we create the InferenceSession which does kernel checking and determines that it does not have the required kernel to execute the adamwoptimizer node on CPU and errors out. We can work around the problem by setting the execution provider when that API is ready.

For now, I disabled it.